unit-tests/src/main/scala/scala/scalanative/native/InstanceOfSuite.scala

```diff
@@ -0,0 +1,79 @@
+package scala.scalanative.native
+
+object InstanceOfSuite extends tests.Suite {
+
+  // asInstanceOf
+
+//  test("expects anyRef.asInstanceOf[String] fail") {
```

Comment:
now `(new AnyRef).asInstanceOf[String]` succeeds
what should be the result here?
is it success in case of prod mode and failure in case of dev mode?